### DISCUSSION TITLE
Move From quasar 1 to Quasar 2

### DISCUSSION THREAD

**Discussion by gasymov-rustam** [quasarframework/quasar#16075] [General - Components / Directives / etc]
Hi guys, i need help, in project was used quasar 1 and vue old version, right now i want to move to newest vesrsion of quasar and vue, and after starting i have some problems, like this

export 'filter' (imported as 'Vue') was not found in 'vue' (possible exports: BaseTransition, BaseTransitionPropsValidators, Comment, EffectScope, Fragment, KeepAlive, ReactiveEffect, Static, Suspense, Teleport, Text, Transition, TransitionGroup, VueElement, assertNumber, callWithAsyncErrorHandling, callWithErrorHandling, camelize, capitalize, cloneVNode, compatUtils, compile, computed, createApp, createBlock, createCommentVNode, createElementBlock, createElementVNode, createHydrationRenderer, createPropsRestProxy, createRenderer, createSSRApp, createSlots, createStaticVNode, createTextVNode, createVNode, customRef, defineAsyncComponent, defineComponent, defineCustomElement, defineEmits, defineExpose, defineModel, defineOptions, defineProps, defineSSRCustomElement, defineSlots, devtools, effect, effectScope, getCurrentInstance, getCurrentScope, getTransitionRawChildren, guardReactiveProps, h, handleError, hasInjectionContext, hydrate, initCustomFormatter, initDirectivesForSSR, inject, isMemoSame, isProxy, isReactive, isReadonly, isRef, isRuntimeOnly, isShallow, isVNode, markRaw, mergeDefaults, mergeModels, mergeProps, nextTick, normalizeClass, normalizeProps, normalizeStyle, onActivated, onBeforeMount, onBeforeUnmount, onBeforeUpdate, onDeactivated, onErrorCaptured, onMounted, onRenderTracked, onRenderTriggered, onScopeDispose, onServerPrefetch, onUnmounted, onUpdated, openBlock, popScopeId, provide, proxyRefs, pushScopeId, queuePostFlushCb, reactive, readonly, ref, registerRuntimeCompiler, render, renderList, renderSlot, resolveComponent, resolveDirective, resolveDynamicComponent, resolveFilter, resolveTransitionHooks, setBlockTracking, setDevtoolsHook, setTransitionHooks, shallowReactive, shallowReadonly, shallowRef, ssrContextKey, ssrUtils, stop, toDisplayString, toHandlerKey, toHandlers, toRaw, toRef, toRefs, toValue, transformVNodeArgs, triggerRef, unref, useAttrs, useCssModule, useCssVars, useModel, useSSRContext, useSlots, useTransitionState, vModelCheckbox, vModelDynamic, vModelRadio, vModelSelect, vModelText, vShow, version, warn, watch, watchEffect, watchPostEffect, watchSyncEffect, withAsyncContext, withCtx, withDefaults, withDirectives, withKeys, withMemo, withModifiers, withScopeId)

 App •  WARNING  •  UI  in ./src/router/index.js

export 'component' (imported as 'Vue') was not found in 'vue' (possible exports: BaseTransition, BaseTransitionPropsValidators, Comment, EffectScope, Fragment, KeepAlive, ReactiveEffect, Static, Suspense, Teleport, Text, Transition, TransitionGroup, VueElement, assertNumber, callWithAsyncErrorHandling, callWithErrorHandling, camelize, capitalize, cloneVNode, compatUtils, compile, computed, createApp, createBlock, createCommentVNode, createElementBlock, createElementVNode, createHydrationRenderer, createPropsRestProxy, createRenderer, createSSRApp, createSlots, createStaticVNode, createTextVNode, createVNode, customRef, defineAsyncComponent, defineComponent, defineCustomElement, defineEmits, defineExpose, defineModel, defineOptions, defineProps, defineSSRCustomElement, defineSlots, devtools, effect, effectScope, getCurrentInstance, getCurrentScope, getTransitionRawChildren, guardReactiveProps, h, handleError, hasInjectionContext, hydrate, initCustomFormatter, initDirectivesForSSR, inject, isMemoSame, isProxy, isReactive, isReadonly, isRef, isRuntimeOnly, isShallow, isVNode, markRaw, mergeDefaults, mergeModels, mergeProps, nextTick, normalizeClass, normalizeProps, normalizeStyle, onActivated, onBeforeMount, onBeforeUnmount, onBeforeUpdate, onDeactivated, onErrorCaptured, onMounted, onRenderTracked, onRenderTriggered, onScopeDispose, onServerPrefetch, onUnmounted, onUpdated, openBlock, popScopeId, provide, proxyRefs, pushScopeId, queuePostFlushCb, reactive, readonly, ref, registerRuntimeCompiler, render, renderList, renderSlot, resolveComponent, resolveDirective, resolveDynamicComponent, resolveFilter, resolveTransitionHooks, setBlockTracking, setDevtoolsHook, setTransitionHooks, shallowReactive, shallowReadonly, shallowRef, ssrContextKey, ssrUtils, stop, toDisplayString, toHandlerKey, toHandlers, toRaw, toRef, toRefs, toValue, transformVNodeArgs, triggerRef, unref, useAttrs, useCssModule, useCssVars, useModel, useSSRContext, useSlots, useTransitionState, vModelCheckbox, vModelDynamic, vModelRadio, vModelSelect, vModelText, vShow, version, warn, watch, watchEffect, watchPostEffect, watchSyncEffect, withAsyncContext, withCtx, withDefaults, withDirectives, withKeys, withMemo, withModifiers, withScopeId)

 App •  WARNING  •  UI  in ./src/router/index.js

export 'directive' (imported as 'Vue') was not found in 'vue' (possible exports: BaseTransition, BaseTransitionPropsValidators, Comment, EffectScope, Fragment, KeepAlive, ReactiveEffect, Static, Suspense, Teleport, Text, Transition, TransitionGroup, VueElement, assertNumber, callWithAsyncErrorHandling, callWithErrorHandling, camelize, capitalize, cloneVNode, compatUtils, compile, computed, createApp, createBlock, createCommentVNode, createElementBlock, createElementVNode, createHydrationRenderer, createPropsRestProxy, createRenderer, createSSRApp, createSlots, createStaticVNode, createTextVNode, createVNode, customRef, defineAsyncComponent, defineComponent, defineCustomElement, defineEmits, defineExpose, defineModel, defineOptions, defineProps, defineSSRCustomElement, defineSlots, devtools, effect, effectScope, getCurrentInstance, getCurrentScope, getTransitionRawChildren, guardReactiveProps, h, handleError, hasInjectionContext, hydrate, initCustomFormatter, initDirectivesForSSR, inject, isMemoSame, isProxy, isReactive, isReadonly, isRef, isRuntimeOnly, isShallow, isVNode, markRaw, mergeDefaults, mergeModels, mergeProps, nextTick, normalizeClass, normalizeProps, normalizeStyle, onActivated, onBeforeMount, onBeforeUnmount, onBeforeUpdate, onDeactivated, onErrorCaptured, onMounted, onRenderTracked, onRenderTriggered, onScopeDispose, onServerPrefetch, onUnmounted, onUpdated, openBlock, popScopeId, provide, proxyRefs, pushScopeId, queuePostFlushCb, reactive, readonly, ref, registerRuntimeCompiler, render, renderList, renderSlot, resolveComponent, resolveDirective, resolveDynamicComponent, resolveFilter, resolveTransitionHooks, setBlockTracking, setDevtoolsHook, setTransitionHooks, shallowReactive, shallowReadonly, shallowRef, ssrContextKey, ssrUtils, stop, toDisplayString, toHandlerKey, toHandlers, toRaw, toRef, toRefs, toValue, transformVNodeArgs, triggerRef, unref, useAttrs, useCssModule, useCssVars, useModel, useSSRContext, useSlots, useTransitionState, vModelCheckbox, vModelDynamic, vModelRadio, vModelSelect, vModelText, vShow, version, warn, watch, watchEffect, watchPostEffect, watchSyncEffect, withAsyncContext, withCtx, withDefaults, withDirectives, withKeys, withMemo, withModifiers, withScopeId)

quasar.config.js
`// Configuration for your app
let path = require("path");
// Get our env variables
const envparser = require("./config/envparser.js");
const get = require("lodash.get");

module.exports = function () {
  return {
    boot: [
      "axios",
      "router-oidc",
      "i18n",
      "oidc",
      "vuelidate",
      "moment",
      "addressbar-color",
      "mask",
      "vue2-google-maps",
      "fragment",
    ],
    css: {
      loaderOptions: {
        css: {
          url: false,
        },
        sass: {
          additionalData: `@import "@/css/app.styl"`,
        },
      },
    },

    extras: ["roboto-font", "material-icons", "fontawesome-v5"],

    framework: {
      components: [
        "QLinearProgress",
        "QLayout",
        "QHeader",
        "QBreadcrumbs",
        "QBreadcrumbsEl",
        "QUploader",
        "QDrawer",
        "QPageContainer",
        "QPage",
        "QPagination",
        "QTooltip",
        "QToolbar",
        "QToolbarTitle",
        "QIcon",
        "QList",
        "QToggle",
        "QInput",
        "QField",
        "QFile",
        "QSelect",
        "QCheckbox",
        "QIntersection",
        "QTime",
        "QDate",
        "QBtn",
        "QBtnDropdown",
        "QBanner",
        "QForm",
        "QItem",
        "QItemSection",
        "QItemLabel",
        "QTabs",
        "QTab",
        "QTabPanels",
        "QTabPanel",
        "QTable",
        "QTh",
        "QTr",
        "QTd",
        "QMarkupTable",
        "QSpace",
        "QRouteTab",
        "QScrollArea",
        "QPageScroller",
        "QSeparator",
        "QCard",
        "QCardSection",
        "QCardActions",
        "QSlideTransition",
        "QChip",
        "QImg",
        "QDialog",
        "QAvatar",
        "QBadge",
        "QPopupProxy",
        "QPopupEdit",
        "QMenu",
        "QInnerLoading",
        "QSpinnerGears",
        "QSpinnerBall",
        "QSpinner",
        "QSpinnerDots",
        "QExpansionItem",
        "QInfiniteScroll",
        "QStepper",
        "QStep",
        "QStepperNavigation",
        "QScrollObserver",
        "QSkeleton",
        "QFab",
        "QFabAction",
        "QOptionGroup",
        "QTree",
        "QRadio",
        "QEditor",
        "QBtnGroup",
      ],

      directives: ["Ripple", "ClosePopup", "Scroll"],

      plugins: [
        "Dialog",
        "Notify",
        "LocalStorage",
        "SessionStorage",
        "Loading",
        "LoadingBar",
        "AddressbarColor",
      ],

      lang: "uk",

      cssAddon: true,
      animations: "all",
      config: {
        screen: {
          bodyClasses: true,
        },
        loadingBar: {
          color: "primary-gradient",
          size: "4px",
          position: "top",
        },
        loading: {
          spinnerColor: "primary",
          spinnerSize: 120,
          backgroundColor: "grey-9",
        },
      },
    },

    supportIE: true,

    build: {
      env: envparser(),
      scopeHoisting: true,
      chainWebpack(cfg) {
        if (process.env.ENVIRONMENT_CONFIGURATION) {
          cfg.output.publicPath("/evo/");
        }
      },
      extendWebpack(cfg) {
        cfg.resolve.alias = {
          ...cfg.resolve.alias,
          "@": path.resolve(__dirname, "./src"),
          "@components": path.resolve(__dirname, "./src/components"),
          "@pages": path.resolve(__dirname, "./src/pages"),
          "@assets": path.resolve(__dirname, "./src/assets"),
        };
        const CopyWebpackPlugin = require("copy-webpack-plugin");
        cfg.plugins.push(
          new CopyWebpackPlugin({
            patterns: [
              {
                from: path.resolve(
                  __dirname,
                  "src/statics/js/OneSignal",
                  "OneSignalSDKWorker.js"
                ),
                to: get(cfg, ["output", "path"], ""),
              },
              {
                from: path.resolve(
                  __dirname,
                  "src/statics/js/OneSignal",
                  "OneSignalSDKUpdaterWorker.js"
                ),
                to: get(cfg, ["output", "path"], ""),
              },
            ],
          })
        );
      },
    },

    devServer: {
      open: true,
    },

    animations: [],

    ssr: {
      pwa: false,
    },

    pwa: {
      manifest: {
        display: "standalone",
        orientation: "portrait",
        background_color: "#ffffff",
        theme_color: "#027be3",
        icons: [
          {
            src: "statics/icons/icon-128x128.png",
            sizes: "128x128",
            type: "image/png",
          },
          {
            src: "statics/icons/icon-192x192.png",
            sizes: "192x192",
            type: "image/png",
          },
          {
            src: "statics/icons/icon-256x256.png",
            sizes: "256x256",
            type: "image/png",
          },
          {
            src: "statics/icons/icon-384x384.png",
            sizes: "384x384",
            type: "image/png",
          },
          {
            src: "statics/icons/icon-512x512.png",
            sizes: "512x512",
            type: "image/png",
          },
        ],
      },
    },

    cordova: {},

    electron: {
      extendWebpack(cfg) {},

      packager: {},

      builder: {},
    },
  };
};
`

package.json
`{
  "name": "e-health",
  "version": "0.0.1",
  "description": "eHealth",
  "productName": "eHealth",
  "cordovaId": "org.cordova.quasar.app",
  "author": "tania.diachuk <tania.diachuk@gmail.com>",
  "private": true,
  "scripts": {
    "lint": "eslint --ext .js,.vue ./",
    "format": "prettier --write \"**/*.{js,vue,scss,html,md,json}\" --ignore-path .gitignore",
    "test": "echo \"No test specified\" && exit 0",
    "start": "quasar dev",
    "start:dev": "quasar dev",
    "clean": "rm -rf node_modules && rm -rf package.lock &&  npm cache clean -f && npm i"
  },
  "dependencies": {
    "@babel/runtime": "7.22.6",
    "@quasar/extras": "^1.16.4",
    "@vue/runtime-dom": "^3.3.4",
    "axios": "^1.4.0",
    "babel-polyfill": "6.26.0",
    "chart.js": "^4.3.0",
    "core-js": "^3.6.5",
    "crypto-js": "^4.1.1",
    "highlight": "^0.2.4",
    "highlight.js": "^11.8.0",
    "inversify": "^6.0.1",
    "lodash.clone": "^4.5.0",
    "lodash.clonedeep": "^4.5.0",
    "lodash.curry": "^4.1.1",
    "lodash.filter": "^4.6.0",
    "lodash.find": "^4.6.0",
    "lodash.first": "^3.0.0",
    "lodash.get": "^4.4.2",
    "lodash.includes": "^4.3.0",
    "lodash.intersectionby": "^4.7.0",
    "lodash.isarray": "^4.0.0",
    "lodash.isempty": "^4.4.0",
    "lodash.isequal": "^4.5.0",
    "lodash.kebabcase": "^4.1.1",
    "lodash.map": "^4.6.0",
    "lodash.mergewith": "^4.6.2",
    "lodash.orderby": "^4.6.0",
    "lodash.pullallwith": "^4.7.0",
    "lodash.remove": "^4.7.0",
    "lodash.set": "^4.3.2",
    "lodash.sortby": "^4.7.0",
    "lodash.transform": "^4.6.0",
    "lodash.union": "^4.6.0",
    "lodash.unionwith": "^4.6.0",
    "lodash.uniq": "^4.5.0",
    "lodash.uniqby": "^4.7.0",
    "moment": "^2.29.4",
    "moment-timezone": "0.5.43",
    "oidc-client": "1.11.5",
    "powerbi-client": "^2.22.3",
    "qs": "^6.11.2",
    "quasar": "^2.6.0",
    "reflect-metadata": "^0.1.13",
    "transliteration": "^2.3.5",
    "v-mask": "^2.3.0",
    "vue": "^3.0.0",
    "vue-chartjs": "^5.2.0",
    "vue-doc-preview": "^0.3.2",
    "vue-i18n": "^9.2.2",
    "vue-router": "^4.0.0",
    "vue-the-mask": "^0.11.1",
    "vue-virtual-scroll-list": "^2.3.5",
    "vue2-google-maps": "^0.10.7",
    "vuelidate": "^0.7.7",
    "vuex": "^4.1.0"
  },
  "devDependencies": {
    "@aspnet/signalr": "^1.0.27",
    "@babel/eslint-parser": "^7.13.14",
    "@quasar/app": "^3.3.3",
    "@quasar/app-webpack": "^3.0.0",
    "@quasar/quasar-app-extension-qpdfviewer": "^1.0.0-beta.9",
    "@vue/eslint-config-standard": "^8.0.1",
    "babel-eslint": "^10.0.1",
    "css-loader": "^6.8.1",
    "eslint": "^8.10.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-plugin-vue": "^9.0.0",
    "eslint-webpack-plugin": "^3.1.1",
    "prettier": "^2.5.1",
    "strip-ansi": "=7.1.0",
    "stylus-loader": "^7.1.3"
  },
  "engines": {
    "node": ">= 12.22.1",
    "npm": ">= 6.13.4",
    "yarn": ">= 1.21.1"
  },
  "browserslist": [
    "last 10 Chrome versions",
    "last 10 Firefox versions",
    "last 4 Edge versions",
    "last 7 Safari versions",
    "last 8 Android versions",
    "last 8 ChromeAndroid versions",
    "last 8 FirefoxAndroid versions",
    "last 10 iOS versions",
    "last 5 Opera versions"
  ],
  "main": "index.js",
  "repository": "https://gitlab.com/it4med/mis.front-end.git",
  "license": "MIT"
}
`

I'm not very familiar with Quasar and Vue, and I would appreciate some help.
Have a nice day)))

**Comment by suleimanas** [comment]
https://quasar.dev/start/upgrade-guide#migrate-to-v2-from-v1

https://www.youtube.com/playlist?list=PLFZAa7EupbB6atAt_liopEDvk44wRa00B
